src/ledger/LedgerTxn.cpp

```diff
@@ -271,19 +352,67 @@ LedgerTxn::Impl::commitChild(EntryIterator iter, LedgerTxnConsistency cons)
         for (; (bool)iter; ++iter)
         {
             auto const& key = iter.key();
+
             if (iter.entryExists())
             {
-                mEntry[key] = std::make_shared<LedgerEntry>(iter.entry());
-            }
-            else if (!mParent.getNewestVersion(key))
-            { // Created in this LedgerTxn
-                mEntry.erase(key);
+                updateEntry(key, std::make_shared<LedgerEntry>(iter.entry()));
             }
             else
-            { // Existed in a previous LedgerTxn
-                mEntry[key] = nullptr;
+            {
+                updateEntry(key, nullptr);
             }
         }
+
+        // We will show that the following update procedure leaves the parent
+        // worst best offer map in a correct state.
+        //
+        // Fix an asset pair P in the child worst best offer map, and let V be
```

Comment:
It would be clearer/shorter to use terminology in the context of "this" as "parent" can be confusing.

Maybe use the proposed notation/definition that I proposed too?

```
        // By definition, child shadows all offers <=? V from its parents (including
        // this LedgerTxn), as a consequence:
        // (1) After the commit, this LedgerTxn will shadow all offers <=? V
        //     from its parents.
        // We need to update the worst offer map to reflect this fact.
        //
        // There are three possibilities prior to the commit:
        // 1. P is not in this LedgerTxn worst best offer map. Because of (1),
        //    V can be associated to this LedgerTxn worst best offer map
        // 2. P is in the parent worst best offer map with associated value W,
        //    and W <=? V. ...
```

Reply:
Updated to use better notation, and also switched the naming convention to "parent", "self", and "child" where child is being committed into self.